class-tgm-plugin-activation.php

```diff
@@ -2675,7 +2675,7 @@ public function process_bulk_actions() {
 					}
 
 					// For updates: make sure this is a plugin we *can* update (update available and WP version ok).
-					if ( 'update' === $install_type && ( $this->tgmpa->is_plugin_installed( $slug ) && ( false === $this->tgmpa->does_plugin_have_update( $slug ) || ! $this->tgmpa->can_plugin_update( $slug ) ) ) ) {
+					if ( 'update' === $install_type && ( ! $this->tgmpa->is_plugin_installed( $slug ) || ( $this->tgmpa->is_plugin_installed( $slug ) && ( false === $this->tgmpa->does_plugin_have_update( $slug ) || ! $this->tgmpa->can_plugin_update( $slug ) ) ) ) ) {
```

Comment:
This set of conditions is too long. It's not easy to read or understand. Please extract into a new documented method.